src/scm/Model.ts

```diff
@@ -347,6 +320,43 @@ export class Model implements Disposable {
         });
 
         this._onDidChange.fire(this.ResourceGroups);
+    }
+
+    private async getDepotOpenedFilePaths(): Promise<string[]> {
+        const output = await Utils.getOutput('opened');
+        const opened = output.trim().split('\n');
+        if (opened.length === 0) {
+            return;
+        }
+
+        const files = [];
+        opened.forEach(open => {
+            const matches = open.match(/(.+)#(\d+)\s-\s([\w\/]+)\s(default\schange|change\s\d+)\s\((\w+)\)/);
+            if (matches) {
+                files.push(matches[1]);
+            }
+        });
+
+        return files;
+    }
 
+    private async getFstatInfoForFiles(files: string[]): Promise<any> {
+        const fstatOutput: string = await Utils.getOutput(`fstat ${files.join(' ')}`);
+        // Windows will have lines end with \n\r.
+        // Each file has multiple lines of output separated by a blank line.
+        // Splitting on \n\r?\n will separate the output for each file.
+        const fstatFiles = fstatOutput.trim().split(/\n\r?\n/);
+        return fstatFiles.map((file) => {
+            const lines = file.split('\n');
+            const lineMap = {};
+            lines.forEach(line => {
+                // ... Name Value
+                const matches = line.match(/[.]{3} (\w+) (.+)/);
```

Comment:
It's not being used right now, but eventually I hope to save all fields, and contextualize commands if the file is mapped, shelved, etc...

see [P4 fstat](https://www.perforce.com/perforce/r14.2/manuals/cmdref/p4_fstat.html) for more info